### DISCUSSION TITLE
Using NFS as PVC for workers

### DISCUSSION THREAD

**Discussion by jdziat** [concourse/concourse#7067] [Help & Support]
Is it possible to use an NFS server for the concourse-worker persistent volume claim in kubernetes? I was hoping it would remove the need for syncing the build cache amongst workers.

**Comment by taylorsilva** [comment]
This would probably not go well. First, the entire Concourse system is designed to assume each worker has its own volume. I'm assuming your NFS solution would mean all workers are sharing the same mount point in your NFS? Second, I don't know how the [worker's baggageclaim `driver`](https://github.com/concourse/baggageclaim/blob/cc2c3140310612ea2c5d3503a3a781d9b0be9228/baggageclaimcmd/command.go#L34) would work with an NFS volume. We recommend using `overlay` as the driver. `naive` would definitely work but would be very slow for ask task initialization.

**Reply by jdziat** [reply]
Shouldn't hurt to try though. Thoughts on: `driver = naive` and `disable-user-namespaces = true`

**Reply by taylorsilva** [reply]
Yeah totally worth a shot! Will be awesome if you can share your results here too for others to find.

So the `naive` driver is quite slow which is why we don't recommend it. You'll notice it the most on large volumes being passed between steps, like the volume generated from a `get` step as the input to a `task` step.
The reason `naive` is slow is because it literally `cp`'s entire folders: https://github.com/concourse/baggageclaim/blob/cc2c3140310612ea2c5d3503a3a781d9b0be9228/volume/driver/naive.go#L24

`disable-user-namespaces` is off by default because from a security perspective, having containers in their own user-namespace is the most secure way of running containers. Plenty of container escape bugs have been found where the workaround is usually "put containers in their own user namespace". I would recommend **not** setting `disable-user-namespaces` to `true`.

**Reply by jdziat** [reply]
Alright makes sense. If I can get it to work i'll report back with metrics across the different drivers/config setups. Thanks for the feedback.
